tests/integration/security/pass_through_filter_chain_test.go

```diff
@@ -39,240 +37,308 @@ func TestPassThroughFilterChain(t *testing.T) {
+							want:   true,
+						},
+						{
+							port:   8088,
+							schema: protocol.TCP,
+							want:   false,
+						},
 					},
-					{
-						target: e,
-						port:   8088,
-						schema: protocol.TCP,
-						want:   false,
-					},
-
-					// For workload f, there is only authN policy that enables mTLS by default and disables mTLS strict on port 8086 and 8088.
+				},
+				{
+					// There is only authN policy that enables mTLS by default and disables mTLS strict on port 8086 and 8088.
 					// The request should be denied on port 8085 and 8071.
-					{
-						target: f,
-						port:   8085,
-						schema: protocol.HTTP,
-						want:   false,
-					},
-					{
-						target: f,
-						port:   8086,
-						schema: protocol.HTTP,
-						want:   true,
-					},
-					{
-						target: f,
-						port:   8087,
-						schema: protocol.TCP,
-						want:   false,
-					},
-					{
-						target: f,
-						port:   8088,
-						schema: protocol.TCP,
-						want:   true,
+					"STRICT with disable",
+					`apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: mtls
+spec:
+  selector:
+    matchLabels:
+      app: a
+  mtls:
+    mode: STRICT
+  portLevelMtls:
+    8086:
+      mode: DISABLE
+    8088:
+      mode: DISABLE`,
+					[]expect{
+						{
+							port:   8085,
+							schema: protocol.HTTP,
+							want:   false,
+						},
+						{
+							port:   8086,
+							schema: protocol.HTTP,
+							want:   true,
+						},
+						{
+							port:   8087,
+							schema: protocol.TCP,
+							want:   false,
+						},
+						{
+							port:   8088,
+							schema: protocol.TCP,
+							want:   true,
+						},
 					},
-				}
-				ctx.NewSubTest(fmt.Sprintf("In %s", cluster.StableName())).Run(func(ctx framework.TestContext) {
-					for _, tc := range cases {
-						name := fmt.Sprintf("A->%s:%d[%t]", tc.target.Config().Service, tc.port, tc.want)
-						a := apps.A.Match(echo.InCluster(cluster)).GetOrFail(ctx, echo.Namespace(ns.Name()))
-						from := getWorkload(a, t)
-						// The request should be handled by the pass through filter chain.
-						host := fmt.Sprintf("%s:%d", getWorkload(tc.target, t).Address(), tc.port)
-						request := &epb.ForwardEchoRequest{
-							Url:     fmt.Sprintf("%s://%s", tc.schema, host),
-							Message: "HelloWorld",
-							Headers: []*epb.Header{
-								{
-									Key:   "Host",
-									Value: host,
+				},
+			}
+
+			for _, cluster := range ctx.Clusters() {
+				client := apps.Naked.Match(echo.InCluster(cluster)).GetOrFail(ctx, echo.Namespace(ns.Name()))
+				destination := apps.A.Match(echo.Namespace(ns.Name())).GetOrFail(ctx, echo.InCluster(cluster))
+				from := getWorkload(client, ctx)
+				destWorkload := getWorkload(destination, ctx).Address()
+				for _, tc := range cases {
+					ctx.NewSubTest(fmt.Sprintf("In %s/%v", cluster.StableName(), tc.name)).Run(func(ctx framework.TestContext) {
+						ctx.Config().ApplyYAMLOrFail(ctx, ns.Name(), tc.config)
```

Comment:
How do you make sure config applied in the last test does not affect the current test case? Same question for the changes to the JWT tests, one benefit of applying configs to multiple instances is that it makes sure any past configs will not affect next tests.

Sometimes the test might be passing due to left-over configs  from last test and thus causing the current test not reflecting the real situation.

Reply:
It is possible. However, we have two mitigations. First, we have `retry.Converge(3)` which means we send multiple requests and ensure they all response the same. The other is we run these tests 100s of times per day, so if we do have a regression and it isn't detected once, it is extremely likely to pop up as a flaky test at least.

We do this for other tests and haven't had issues here, but open to suggestions - @stevenctl ? We could add an `istioctl wait` but its not very reliable

One benefit of this is we are gaining coverage of the envoy config changing, which has had a lot of bugs historically.

Reply:
Other security tests do exactly that:

https://github.com/istio/istio/blob/6d5406628e2fa9bd2fa31e9b4bc5d05186e01bd6/tests/integration/security/util/reachability/context.go#L103

https://github.com/istio/istio/blob/6d5406628e2fa9bd2fa31e9b4bc5d05186e01bd6/pkg/test/framework/components/istioctl/kube.go#L61

Reply:
Can we also use `WaitForConfigWithSleep` here and jwt_test.go? the retry may not always help when the current test passed for the first time due to effect from last test, in this case the retry won't be triggered.

Reply:
By the way, did we use `retry.Converge()` here? I only find the `DefaultCallRetryOptions()`.